enhancements/cluster-logging/elasticsearch-cert-management.md

```diff
@@ -0,0 +1,179 @@
+
+## Motivation
+
+Currently our managed Elasticsearch cluster requires anything using our operator to generate, manage, and provide certs and secrets for the operator and its cluster to use. This leads to many potential sources to create certificates and can cause a gap between what the managed Elasticsearch requires and what something may provide. This would instead make the Elasticsearch Operator a single source of truth for how certificates would be generated.
+
+### Goals
+The specific goals of this proposal are:
+
+* Outline the responsibility for the Elasticsearch Operator regarding certificates
+
+* Determine an annotation that the Elasticsearch Operator will watch for on secrets in the same namespace as an existing elasticsearch CR
+
+* Discuss upgrade strategies from the current LTS mechanism.
+
+We will be successful when:
+
+* Users of Elasticsearch Operator no longer need to maintain certificates for Elasticsearch and can have certificates injected into their secrets
+
+* There is a clear path forward for automatic upgrades
+
+### Non-Goals
+
+* This does not seek to implement work to adhere to the cluster level TLS definitions, however this does make the implementation less complex for the Elasticsearch use-case.
+
+## Proposal
+
+Instead of the current implementation where any consumer of the Elasticsearch Operator is responsible for providing certificates and secrets for Elasticsearch and the operator to use, we want to shift this responsibility to the Elasticsearch Operator instead. This will greatly improve the use case of the operator and make the operator the single source of truth for its certificates.
+
+To allow other components to still be able to communicate with the managed cluster via mTLS the operator will watch for a specific annotation on secrets in the namespace where an elasticsearch CR is present and will inject into this secret.
+
+### User Stories
+
+#### As an user of EO, I do not want to have to generate and provide secrets for the Elasticsearch cluster
+
+We want to further simplify using the Elasticsearch Operator and create a single source of truth for the certs the cluster uses.
+
+#### As a consumer of EO, I want a mechanism to have certificates injected into a secret for me to continue mTLS communcation with my Elasticsearch cluster
+
+We want to continue allowing mTLS as this supports the Cluster Logging and Jaeger use cases, however they shouldn't need to do more than notify they need certs injected by EO.
+
+### Implementation Details
+
+#### Assumptions
+
+* The Elasticsearch Operator will implement this in golang (rather than the current script used in CLO)
+
+* The operator will need to be able to store generated certificates for each namespace/unique Elasticsearch cluster so that there is no security concerns of clusters cross communicating.
+
+#### Security
```

Comment:
Following up here as I don't know if we addressed it.  We need to either express a way for users to request an ES permission set/role or explicitly list what permission/set or role you get when get these certs.  Do we need to be concerned with restricting how can ask for certs or is it sufficient to say if you have the permissions to request then that is restrictive enough

Reply:
> We need to either express a way for users to request an ES permission set/role or explicitly list what permission/set or role you get when get these certs

do we have a requirement for this anywhere? We don't currently offer this functionality

Reply:
I don't know that we do need someone to request a certain permission;  I only put it out there as a possibility.  I think if, with this proposal, we explicitly state the role you will be given when you request certs, I believe that is sufficient and we are covered.

Reply:
I think this would be considered a stretch goal/completely separate proposal... we are currently only looking to move cert management into the operator. This starts to get into the realm of EO dynamically seeding ES.

Reply:
Some way we still need to be explicit about what role will be granted even if it is to state you will get the roles that existed prior to implementation.  We know what they are so we should be able to state them.  My main concern is we are bound to have someone ask how to get some alternate ES permission.  Defining what you get with this proposal provides us a pointer for when someone asks

Reply:
> My main concern is we are bound to have someone ask how to get some alternate ES permission. Defining what you get with this proposal provides us a pointer for when someone asks

We can adjust the proposal if/when that happens